resources/subserver.rb

```diff
@@ -14,20 +14,86 @@
 # limitations under the License.
 #
 
-actions :enable, :disable
+resource_name :aix_subserver
 
-attribute :servicename, name_attribute: true, kind_of: String
-attribute :type, kind_of: String, equal_to: %w(dgram stream sunrpc_udp sunrpc_tcp)
-attribute :protocol, kind_of: String, required: true, equal_to: %w(tcp udp tcp6 udp6)
-attribute :wait, kind_of: String, default: 'nowait', equal_to: %w(wait nowait SRC)
-attribute :user, kind_of: String, default: 'root', required: true
-attribute :program, kind_of: String
-attribute :args, kind_of: String
-
-attr_accessor :enabled
+property :servicename, String, name_property: true, identity: true
+property :type, String, equal_to: %w[dgram stream sunrpc_udp sunrpc_tcp]
+property :protocol, String, required: true, equal_to: %w[tcp udp tcp6 udp6], identity: true
+property :wait, String, equal_to: %w[wait nowait SRC], default: 'nowait'
+property :user, String, required: true, default: 'nowait'
+property :program, String
+property :args, String
+property :enabled, [true, false]
```

Comment:
If you're determining the state of these within the cookbook you need to set desired_state: false